src/bgw/job_stat.c

```diff
@@ -205,6 +205,18 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
 																		  consecutive_failures);
 	MemoryContext oldctx;
+
+	/* stop to execute failing jobs after reached the "max_retries" option */
+	if (job->fd.max_retries > 0 && consecutive_failures >= job->fd.max_retries)
+	{
+		elog(LOG,
+			 "%s: max_retries reached %d, consecutive failures %d",
```

Comment:
Users need to take action on this, if needed. So we should add more info about the job  to the message.
```suggestion
			 "job %d not executed as max_retries reached %d, consecutive failures %d",
```

Reply:
Sure, just forgot to improve the log message. Fixed, thanks!